testing/src/main/java/org/oppia/android/testing/story/StoryProgressTestHelper.kt

```diff
@@ -0,0 +1,774 @@
+import org.oppia.android.domain.topic.RATIOS_EXPLORATION_ID_2
+import org.oppia.android.domain.topic.RATIOS_EXPLORATION_ID_3
+import org.oppia.android.domain.topic.RATIOS_STORY_ID_0
+import org.oppia.android.domain.topic.RATIOS_STORY_ID_1
+import org.oppia.android.domain.topic.RATIOS_TOPIC_ID
+import org.oppia.android.domain.topic.StoryProgressController
+import org.oppia.android.domain.topic.TEST_EXPLORATION_ID_0
+import org.oppia.android.domain.topic.TEST_EXPLORATION_ID_1
+import org.oppia.android.domain.topic.TEST_EXPLORATION_ID_2
+import org.oppia.android.domain.topic.TEST_EXPLORATION_ID_3
+import org.oppia.android.domain.topic.TEST_EXPLORATION_ID_4
+import org.oppia.android.domain.topic.TEST_EXPLORATION_ID_5
+import org.oppia.android.domain.topic.TEST_STORY_ID_0
+import org.oppia.android.domain.topic.TEST_STORY_ID_1
+import org.oppia.android.domain.topic.TEST_STORY_ID_2
+import org.oppia.android.domain.topic.TEST_TOPIC_ID_0
+import org.oppia.android.domain.topic.TEST_TOPIC_ID_1
+import org.oppia.android.testing.TestCoroutineDispatchers
+import org.oppia.android.testing.time.FakeOppiaClock
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private val EIGHT_DAYS_IN_MS = TimeUnit.DAYS.toMillis(8)
+
+/**
+ * This helper allows tests to easily create dummy progress per profile-basis.
+ *
+ * Note that it's required that tests depending on this helper change their fake Oppia clock to use
+ * uptime millis rather than wall clock time & leverage the test coroutine dispatcher's ability to
+ * move forward Robolectric's fake time clock for better reliability when recording chapter progress
+ * or completion.
+ */
+@Singleton
+class StoryProgressTestHelper @Inject constructor(
+  private val storyProgressController: StoryProgressController,
+  private val fakeOppiaClock: FakeOppiaClock,
+  private val testCoroutineDispatchers: TestCoroutineDispatchers
+) {
+  @Mock lateinit var mockLiveDataObserver: Observer<AsyncResult<Any?>>
+  @Captor lateinit var liveDataResultCaptor: ArgumentCaptor<AsyncResult<Any?>>
+
+  init {
+    MockitoAnnotations.initMocks(this)
+  }
+
+  /* Test topic completion methods. */
+
+  /**
+   * Marks the first chapter of test topic 0 story 0 as completed.
+   *
+   * Note that this function will advance the clock & synchronize all background execution to ensure
+   * that any operations needed to achieve the requested state are completed and successful.
+   *
+   * @param profileId the profile we are setting partial progress of the fraction story for
+   * @param timestampOlderThanOneWeek if the timestamp for this topic progress is more than one week
+   *     ago
+   */
+  fun markCompletedTestTopic0Story0Exp2(profileId: ProfileId, timestampOlderThanOneWeek: Boolean) {
```

Comment:
@BenHenning Here you are following the convention of id names which can make it confusing to use this function at other places. 
Considering these functions will be used in other files I think the name should be `markCompletedTestTopic0Story0Exp0` so that it looks less confusing in other files and in future we should update the `ids` too to match these. WDYT?

Reply:
I was on the fence about this since the opposite situation could easily happen (e.g. assuming that TEST_EXPLORATION_ID_0 is marked as completed in this case). Given both situations seem relatively equal, I don't think I have a strong stance.

@vinitamurthi / @anandwana001 / @Srikanth-Kb do any of you have thoughts here?

Reply:
In my experience `markCompletedTestTopic0Story0Exp0` function or other helper functions are generally getting more used as compared to topic ids in app layer and therefore I am particularly in favour of `markCompletedTestTopic0Story0Exp0`

Reply:
Hmm I think this was merged before we could address this. I'll follow up with another PR to fix this @rt4914.

Reply:
Opened: #2695.